### PR TITLE
Improving the duration of the scheduled timer tests.

### PR DESCRIPTION
**Pull request by msft-Jeyaram**
This needs to be verified on a different machine. As I had issue with timer not going off for the specified duration. 
@mnithish is trying it.

### CODE REVIEW

File: tests/unittests/Foundation/NSTimerTests.mm [hunk 1]
```diff
@@ -133,19 +132,18 @@ - (void)dealloc {
...
     volatile long waitCompletion = 0;
 
-    auto selectorCalledAsync = std::async(std::launch::async,
-                                          [&waitCompletion, testObj]() {
-                                              BOOL fooChanged = [testObj waitOnCalledConditionForInterval:15] && [testObj called];
-                                              _InterlockedExchange(&waitCompletion, 1L);
-                                              return fooChanged;
-                                          });
+    auto selectorCalledAsync = std::async(std::launch::async, [&waitCompletion, testObj]() {
+        BOOL fooChanged = [testObj waitOnCalledConditionForInterval:5] && [testObj called];
```
Comment (msft-Jeyaram) [inline tests/unittests/Foundation/NSTimerTests.mm]: I can reduce this to 3/4

**Comment by DHowett-MSFT** [comment]
My only concern with this is that we're making heavy-handed tests more likely to fail if the machine is under stress. Is there a way to write these tests that doesn't require a timed waiter and a timed waitee?
If we were to have a timed waitee and an un-timed waiter (say: holding a mutex, or waiting on a condition), we might be able to avoid stressed systems failing these tests.

**Comment by msft-Jeyaram** [comment]
True, a system under stress can delay the timers. Although we seem to have timed test pattern in some other tests in our code base (e.g NSObject_PerformSelector* tests). I'll look into this.

---
In reply to: [232461993](https://github.com/Microsoft/WinObjC/pull/642#issuecomment-232461993) [](ancestors = 232461993)

**Comment by mnithish** [comment]
I agree with Dustin, I am more than confident these will fail in our nightly runs

---
In reply to: [232463598](https://github.com/Microsoft/WinObjC/pull/642#issuecomment-232463598) [](ancestors = 232463598,232461993)

File: tests/unittests/Foundation/NSTimerTests.mm [hunk 2]
```diff
@@ -105,16 +105,15 @@ - (void)dealloc {
...
+    auto selectorCalledAsync = std::async(std::launch::async, [&waitCompletion, testObj]() {
+        BOOL fooChanged = [testObj waitOnCalledConditionForInterval:2] && [testObj called];
+        _InterlockedExchange(&waitCompletion, 1L);
+        return fooChanged;
+    });
 
     NSRunLoop* runLoop = [NSRunLoop currentRunLoop];
 
-    NSTimer* timer = [NSTimer scheduledTimerWithTimeInterval:10 target:testObj selector:@selector(testFunction) userInfo:nil repeats:NO];
+    NSTimer* timer = [NSTimer scheduledTimerWithTimeInterval:1 target:testObj selector:@selector(testFunction) userInfo:nil repeats:NO];
```
Comment (MSFTFox) [inline tests/unittests/Foundation/NSTimerTests.mm]: Might be worth it to put some of these time intervals and other numbers at the top as statics to make updating this test easier in the future as well.

File: tests/unittests/Foundation/NSTimerTests.mm [hunk 3]
```diff
@@ -105,16 +105,15 @@ - (void)dealloc {
...
     volatile long waitCompletion = 0;
 
-    auto selectorCalledAsync = std::async(std::launch::async,
-                                          [&waitCompletion, testObj]() {
-                                              BOOL fooChanged = [testObj waitOnCalledConditionForInterval:15] && [testObj called];
-                                              _InterlockedExchange(&waitCompletion, 1L);
-                                              return fooChanged;
-                                          });
+    auto selectorCalledAsync = std::async(std::launch::async, [&waitCompletion, testObj]() {
+        BOOL fooChanged = [testObj waitOnCalledConditionForInterval:2] && [testObj called];
```
Comment (bbowman) [inline tests/unittests/Foundation/NSTimerTests.mm]: >2 [](start = 68, length = 1)

in terms of unit tests 2 seconds is still a long time. Perhaps consider something in the ms range.

**Comment by bbowman** [comment]
>     }

busy waits are about my least favorite mechanism to do multithreaded tests. Especially since this is hard to get correct as coded. I forget all of the details on volatile and would much prefer us to use a condition variable, std::atomic or other syncronization object.

---
Refers to: tests/unittests/Foundation/NSTimerTests.mm:128 in db05926. [](commit_id = db059265324b4921a4a77bb73b9f324f278f4bb4, deletion_comment = False)

**Comment by bbowman** [comment]
:clock1:

**Comment by rajsesh-msft** [comment]
Is anything happening with this PR?

**Comment by msft-Jeyaram** [comment]
this will be obsolete with the use of CFRunLoopTimer as most of the time based tests are removed to use conditional checks.

**Comment by rajsesh-msft** [comment]
Thanks @msft-Jeyaram in that case, please close this pull request.

**Pull request closed without merging**